### ISSUE TITLE
Qt6 qtbase build success is VERY hit or miss

### ISSUE THREAD

**Issue by Schala** [microsoft/vcpkg#18427]
**Host Environment**
- OS: Windows 10
- Compiler: MSVC 16.10.1

**To Reproduce**
`vcpkg install qtbase`

**Failure logs** 
`-- Using cached C:/vcpkg/downloads/qtbase-v6.1.0.tar.gz
-- Extracting source C:/vcpkg/downloads/qtbase-v6.1.0.tar.gz
-- Applying patch jpeg.patch
-- Applying patch harfbuzz.patch
-- Applying patch config_install.patch
-- Applying patch allow_outside_prefix.patch
-- Applying patch buildcmake.patch
-- Applying patch dont_force_cmakecache.patch
-- Applying patch fix_find_dep.patch
-- Using source at C:/vcpkg/buildtrees/qtbase/src/v6.1.0-fd453347c4.clean
-- Found external ninja('1.10.2').
-- Configuring x64-windows
-- Building x64-windows-dbg
-- Building x64-windows-rel
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:146 (message):
    Command failed: "C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/bin/cmake.exe" --build . --config Release --target install -- -v -j17
    Working Directory: C:/vcpkg/buildtrees/qtbase/x64-windows-rel
    See logs for more information:
      C:\vcpkg\buildtrees\qtbase\install-x64-windows-rel-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:104 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:44 (vcpkg_build_cmake)
  ports/qtbase/cmake/qt_install_submodule.cmake:112 (vcpkg_install_cmake)
  ports/qtbase/portfile.cmake:208 (qt_install_submodule)
  scripts/ports.cmake:139 (include)
`

**Additional context**
For some reason, building the new Qt6 qtbase port is a huge hassle. This might apply to other Qt6 packages as well. When building, the build process ends up doing one of two things: CMake just sits there doing nothing after a while (I had the same _CL* files in my %tmp% folder for a good half hour before giving up, no new ones were being generated either) or it building the debug build but fails on the release build.

**Comment by Neumann-A** [comment]
CI disagrees (It basically builds it 6 times on different platforms for every change to a dependency). 
I build it numerous times without any problem (Including GitHub action).  
Maybe AV is screwing you? You can also try to deactivate ninja by removing `PREFER_NINJA` from https://github.com/microsoft/vcpkg/blob/92b42c4c680defe94f1665a847d04ded890f372e/ports/qtbase/cmake/qt_install_submodule.cmake#L82. or passing `DISABLE_NINJA` In the portfile

**Comment by Schala** [comment]
Now that you mention it, when I checked task manager, "Antimalware Service Executable" was eating up about 3 GB of memory. I wonder if I can tell it to take a hike while I build Qt6?

**Comment by Osyotr** [comment]
@Schala yes you can. It's also a good idea to exclude vcpkg.exe, cmake.exe, ninja.exe, cl.exe, ld.exe, devenv.exe and other related processes from scan. Build directories too.
And check your free storage space.

**Comment by Schala** [comment]
Thanks. I have 16 GB of RAM and was experiencing no slow downs, yet still was experiencing this. Tomorrow I'll plug those into the Antimalware and try it out.

**Comment by Schala** [comment]
Ah sorry, forgot to close this. I excluded vcpkg, cmake and ninja from Antimalware, and this has been running fine. Thanks for the hint!
